### DISCUSSION TITLE
Different leastsq results with new numpy version

### DISCUSSION THREAD

**Discussion by mbakker7** [lmfit/lmfit-py#981] [General]
I only upgraded numpy (from 1.x to 2.x) and now I get different results for lmfit. 
Just doing a minimization with leastsq. 
Has anybody seen that before?

**Comment by newville** [comment]
@mbakker7 Hm, I have not noticed that, but I think we have not done a careful comparison.  Like, I think we were just happy that numpy2 worked!  Our tests pass, but these tend not to be too fussy about the precision of the results.   

I am not sure what from numpy would be numerically very different.  OTOH, I would not be shocked at changes with scipy versions.  

Can you post a simple example that shows a difference?  Thanks.

**Reply by mbakker7** [reply]
A simple example is not as easy as a more difficult example. Getting slightly different results with a simple example (i.e., 4th digit kind of stuff).
What about changes in scipy version? Does that make a bit difference? What version of scipy should I use?

**Comment by newville** [comment]
@mbakker7 I can definitely believe scipy versions could have an impact, especially for some solvers, though I do not know of anything in particular for any recent scipy versions (like, over several years).   I usually just use "latest available", but if you're looking for differences between versions, I would believe changes in lmfit version, then scipy version, then numpy.
